db/migrate/20180426145669_migrate_tax_categories_to_line_items.spree.rb

```diff
@@ -1,8 +1,12 @@
 # This migration comes from spree (originally 20130802022321)
 class MigrateTaxCategoriesToLineItems < ActiveRecord::Migration
   def change
-  	Spree::LineItem.includes(:variant => { :product => :tax_category }).find_in_batches do |line_items|
-  	  line_items.each do |line_item|
+    Spree::LineItem.includes(:variant => { :product => :tax_category }).find_in_batches do |line_items|
+      line_items.each do |line_item|
+        next if line_item.variant.nil?
+        next if line_item.variant.product.nil?
+        next if line_item.product.nil?
+        next unless line_item.product.tax_category.present?
```

Comment:
Both variants and tax categories have default scope (both use `deleted_at IS NULL`, and tax category has `"spree_tax_categories"."is_default" = 't'`). These should be unscoped in the checks above - the DB still knows the tax category even if the variant and tax category have already been soft-deleted.

Reply:
This code is copied from Spree.
I dont see much benefit in updating the tax category of a line item with a variant that has been deleted...
I think I am saying this because I dont know how to do it 🤣 

Is there a way to go through the associations like line_item.variant.unscoped?

Reply:
Yes, there is `unscoped` but it would return all variants, not just the ones of that line item. There should be a special scope `with_deleted` for this case though.

Reply:
> I dont see much benefit in updating the tax category of a line item with a variant that has been deleted...

Thanks, @luisramos0. I'm okay with keeping this as is then. @mkllnk Are you okay with not touching this anymore too?

Reply:
Yes, all good.